integration_tests/sdk/data_integration_tests/s3_test.py

```diff
@@ -422,6 +422,10 @@ def test_s3_basic_fetch_failure(client, data_integration):
     with pytest.raises(AqueductError, match="The specified key does not exist."):
         data_integration.file("asdlkf", artifact_type=ArtifactType.TABLE, format="parquet")
 
+    # Fetch a path to directory that does not exist will fail.
+    with pytest.raises(AqueductError, match="Given path to S3 directory 'asdlkf/' does not exist."):
+        data_integration.file("asdlkf/", artifact_type="bytes")
+
```

Comment:
Do we have a test case for an empty directory? If not we should add it.

Reply:
I think this piece of testing might be a little bit tricky to setup. If I understand correctly, we set up S3 in `setup_integration.py` by publishing a "save" flow that save all tables from demo_db to s3 storage. However, in order to create a folder, we need a s3 client which I think has been abstracted away in `aqueduct executor`. 

A way I think we can do it is create a S3 client using config info inside `setup_integration.py` and use that to achieve the purpose. I am not sure if that's ok within the testing?

Reply:
Actually I think we can write the following test case:
```
1) Create an object at some_dir/object
2) Delete the workflow, and include that object in the deletion.
3) some_dir/ should now be an empty directory.
```
This will also help us clarify what our deletion logic too.

Reply:
I have included that into the test but there are some discrepancies in test coverage:
 
Behavior 1(Integration Test):
- save an object to path `new_directory\obj_name`, publish the workflow, delete the flow with saved object
- The object is deleted as well as the directory, which we prompted to user `directory does not exist..`

Behavior 2(tried that on jupyter):
- create an empty folder on S3 console with name `existing_directory`
- save an object to path `existing_directory\obj_name`, publish the workflow, delete the flow with saved object
- The directory exists and is an empty directory

I think this test is still good to include but it seems like there is a difference between console created folder and folder created by creating objects.

Reply:
Oh hmm that is kind of unexpected... why are we deleting the parent directory? @cw75 do you know?